src/processTargets/modifiers/scopeTypeStages/RegexStage.ts

```diff
@@ -82,11 +77,7 @@ class RegexStage implements ModifierStage {
       range.contains(position)
     );
     if (match == null) {
-      if (this.name) {
-        throw new Error(`Couldn't find containing ${this.name}`);
-      } else {
-        throw new Error(`Cannot find sequence defined by regex: ${this.regex}`);
-      }
+      throw new NoContainingScopeError(this.modifier.scopeType.type);
```

Comment:
What if we just passed in the whole `scopeType` to `NoContainingScopeError`, and then we have a function that converts a scope type to text when we display the error message.  So eg for simple scope types it just converts camel case to regular text (can just use [`lodash.lowerCase`](https://lodash.com/docs/4.17.15#lowerCase), and for regex scope types it prints out `regex ${regex}`

Reply:
I would be fine with that. But I'm not sure if it fits in this pr

Reply:
I guess the main issue I'd like to avoid is the blanket error catching below in custom regex; the approach I proposed strikes me as the right approach, but if you can find something simpler, I guess we can consider the above out of scope 🤷.  Seems related enough to fit into this PR, but I don't feel strongly

Reply:
I don't see how this is worse than what was there before?

Reply:
The problem is the blanket error catching. So something else could go wrong and we report it as couldn't find scope type

Reply:
If something else goes wrong we should not use that error. In this case we only throw this error if we can't find a match.  I don't quite follow the problem here?